### PR TITLE
Adds new AuditLog trait to daphne

### PR DESCRIPTION
**Pull request by will118**
Enables implementations backed by services like: https://developers.cloudflare.com/analytics/analytics-engine/

* For now, there is just a basic handler for successful aggregation job requests.
* Adds a `Display` to `VdafConfig` for a reasonably succinct string repr, useful for event logging.
* Adds some assertions on the handler being invoked

### CODE REVIEW

**Comment by mendess** [comment]
Isn't it possible to leverage tracing somehow to achieve this? Adding this effectively means we have to visibility mechanisms

File: daphne_worker/Cargo.toml
Comment (cjpatton) [inline daphne_worker/Cargo.toml:43]: Is bumping the `worker` crate necessary in order to use new features? If not, then I'd prefer to bump the version in a separate PR. The reason is that in the past we've seen bugs in `worker` releases that only manifested in the real Workers runtime (not when run locally via `wrangler dev`). Ideally we would test updates to `worker` in a live deployment before merging.

File: daphne/src/audit_log.rs
Comment (cjpatton) [inline daphne/src/audit_log.rs:1]: New files must have copyright notices.

File: daphne/src/audit_log.rs
Comment (cjpatton) [inline daphne/src/audit_log.rs:4]: Why do we need a separate callback for agg init and and continue? How about a single callback for aggregate requests?

File: daphne/src/audit_log.rs
Comment (cjpatton) [inline daphne/src/audit_log.rs:2]: Can you add a docucomment briefly describing what this does? IIUC the implementation of the trait is in the internal repo for now, but it would be good to at least to say what this does and provide a link to internal documentation.

File: daphne/src/lib.rs
Comment (cjpatton) [inline daphne/src/lib.rs:722]: nit: For consistency with match on `VdafConfig` elsewhere 
```suggestion
            VdafConfig::Prio3(prio3_config) => write!(f, "Prio3({conf})"),
```

File: daphne/src/roles.rs
Comment (cjpatton) [inline daphne/src/roles.rs:1047]: Without knowing what we're doing with this information, I don't quite see the value of having two callbacks, one for agg init and one for agg cont. It seems like the same information is being reported.

File: daphne_worker/Cargo.toml
Comment (will118) [reply daphne_worker/Cargo.toml:43]: we can do it in a separate PR :+1:

File: daphne/src/audit_log.rs
Comment (will118) [reply daphne/src/audit_log.rs:4]: we don't, I can update it to take a boolean if that is preferred.

File: daphne/src/audit_log.rs
Comment (will118) [reply daphne/src/audit_log.rs:2]: yep, I can do so after we merge this into the fork

File: daphne/src/audit_log.rs
Comment (mendess) [reply daphne/src/audit_log.rs:4]: I'd rather keep boolean parameters to a minimum, they make code hard to read. You can instead define a simple two variant enum, same performance, much better readability
```rs
enum AuditAction {
    Init,
    Continue,
}
```

**Review by mendess — APPROVED** [review]
Approving to unblock this, we can iterate on using tracing for this at a latter date

**Pull request merged**
